tests/providers/conftest.py

```diff
@@ -31,6 +31,8 @@ def disconnected_provider(request):
         return provider
     elif request.param == 'rpc':
         return RPCProvider(port=9999)
+    elif request.param == 'rpc':
+        return KeepAliveRPCProvider(port=9999)
```

Comment:
I don't think this is going to work with assigning the same port.  I believe it would be best to use the `get_open_port` function (feel free to patch the previous hard-coded port as well)